### DISCUSSION TITLE
Require is not defined

### DISCUSSION THREAD

**Discussion by Robovize** [ujjwalguptaofficial/JsStore#267] [Q&A]
Hello, I try to use JsStore in Vue.js, and when I tried to copy/paste the content of files from https://github.com/ujjwalguptaofficial/jsstore-examples/tree/master/vue  into my test project I get an error

require is not defined

![image](https://user-images.githubusercontent.com/56894839/167169472-ecd6e27b-b2dd-4513-9cd9-6a8981faa18b.png)

and the error is coming from line 6 
![image](https://user-images.githubusercontent.com/56894839/167169594-8717d332-b280-4954-b98f-a4b85b391db0.png)

I create a new Vue.js project with Vite and then just copy-paste the content of .vue files and .js files. 

Am I missing something?

**Comment by ujjwalguptaofficial** [comment]
I guess vite is using rollup and that syntax is particularly for webpack, you gotta find how to use web worker in rollup.

or

 you can use without web worker syntax

**Reply by john-easci** [reply]
Well, you can't.  The import itself is doing an old-school require which is breaking. 

Even with the commonjs plugin, the webpack setup still breaks after trying to hook directly into the JsStore export.

I, for one, am moving on to a more modern solution.

**Reply by supermanZhouzihan** [reply]
I also get this error ,but I use vue-cli to create vue2 project,
use npm to install jsstore ,run my app give me tips need file-loader and I  install  ，and run app,get this error
![1685849998185(1)](https://github.com/ujjwalguptaofficial/JsStore/assets/42661601/b2ec76d4-2bc7-4c40-8db3-4340e6cd8d06)
![1685850676454](https://github.com/ujjwalguptaofficial/JsStore/assets/42661601/8f46adf9-f60a-435f-a59e-33453efa7cd3)

**Reply by supermanZhouzihan** [reply]
I Found the reason 
change transpileDependencies to false
